notebooks/official/tabular_workflows/tabnet_on_vertex_pipelines.ipynb

```diff
@@ -0,0 +1,1251 @@
+{
```

Comment:
<p>Could you please clarify why some of the parameters are set by overriding the study_spec and others are set directly within the get_builtin_algorithm_hyperparameter_tuning_job_pipeline_and_parameters function call?</p><p> </p><p>Is there anything we can do to improve the appearance of the auto-generated API on https://google-cloud-pipeline-components.readthedocs.io/? The pages are not displaying properly for me. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/726/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='48'/>

Reply:
Any hyperparameter that can be tuned is set as part of study_spec_parameters_override and anything that cannot (ex. frequency at which to run evaluation) is set in the function directly. 

Yeah, the docs don't look the best (especially get_builtin_algorithm_hyperparameter_tuning_job_pipeline_and_parameters). I will looked into how readthedocs.io formatting works and see if there's anything we can do.

Reply:
<p>Ok, so let's say something like this: </p><p> </p><p>To get the best set of hyperparameters for your dataset, we recommend running a HyperparameterTuningJob. </p><p> </p><p>Hyperparameters that can be tuned are set in the 'study_spec_parameters_override' parameter. The tuning is performed using a helper function called  'get_tabnet_study_spec_parameters_override'. You provide <code>dataset_size_bucket</code> (one of 'small', 'medium' and 'large'), <code>training_budget_bucket</code> (one of 'small', 'medium' and 'large'), and <code>prediction_type</code> and Vertex AI recommends a set of hyperparameters and ranges. For a full list of hyperparameters available for tuning, see []. </p><p> </p><p>Hyperparameters that cannot be tuned are set directly in the function call of the HyperparameterTuningJob. In the example below, we configure the following hyperparameters:</p><ul><li>max_trial_count: ...</li><li>parallel_trial_count: ...</li><li>algorithm: ...</li><li>max_failed_trial_count: ...</li><li>study_spec_algorithm: ...</li></ul><p> </p><p>In addition to hyperparameters, HyperparameterTuningJob takes the following values in the example below:</p><ul><li>root_dir: ...</li><li>...</li></ul><p> </p><p>For a full list of HyperparameterTuningJob parameters, see []</p>

Reply:
Ok, I reworked the section to more match that format. Two differences:

1. max_trial_count, parallel_trial_count etc. aren't hyperparameters, they're inputs to the HyperparameterTuningJob so I included them in the last section (In addition to hyperparameters,..)
2. `study_spec_parameters_override` is actually optional. The helper function returns all the hyperparameters to let the customer know what can be tuned, but none, some, or all of them can be specified. We'll fill in ranges for hyperparameters that aren't specified in the override in the pipeline. Tried to clarify that in the text. Please let me know if you think it should be reworded.